var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -223,106 +230,100 @@ def libs(self):
             libraries, root=self.prefix, shared=shared, recursive=True
         )
 
-    @run_before('configure')
     def fortran_check(self):
         if '+fortran' in self.spec and not self.compiler.fc:
             msg = 'cannot build a Fortran variant without a Fortran compiler'
             raise RuntimeError(msg)
 
-    def configure_args(self):
+    def java_check(self):
+        if '+java' in self.spec and not self.compiler.java:
+            msg = 'cannot build a Java variant without a Java compiler'
+            raise RuntimeError(msg)
+
+    def cmake_args(self):
+        args = [
+            '-DBUILD_SHARED_LIBS={0}'.format(
+                'ON' if '+shared' in self.spec else 'OFF')
```

Comment:
Prefer for readability and robustness against variant typos:
```python
args = [self.define_from_variant('BUILD_SHARED_LIBS', 'shared')]
```

Reply:
Done